tensorboard/webapp/routes/feature_flag_serializer.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Store} from '@ngrx/store';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+import {Location} from '../app_routing/location';
+import {SerializableQueryParams} from '../app_routing/types';
+import {State} from '../app_state';
+import * as selectors from '../selectors';
+import {EXPERIMENTAL_PLUGIN_QUERY_PARAM_KEY} from '../webapp_data_source/tb_feature_flag_data_source_types';
+import {FeatureFlagMetadata, FeatureFlagQueryParameters,} from '../webapp_data_source/tb_feature_flag_query_parameters';
+
+export function getFeatureFlagStates(store: Store<State>):
+    Observable<SerializableQueryParams> {
+  return store.select(selectors.getEnabledExperimentalPlugins)
+      .pipe(map((experimentalPlugins) => {
+        const queryParams = experimentalPlugins.map((pluginId) => {
+          return {key: EXPERIMENTAL_PLUGIN_QUERY_PARAM_KEY, value: pluginId};
+        });
+
+        const currentQueryParams = Object.fromEntries(
+            serializableQueryParamsToEntries(new Location().getSearch() || []));
+
+        Object.values(FeatureFlagQueryParameters)
+            .forEach((overriddenFeatureFlag: FeatureFlagMetadata) => {
+              const queryParamOverride =
+                  overriddenFeatureFlag.queryParamOverride;
+              if (queryParamOverride &&
+                  queryParamOverride in currentQueryParams) {
+                queryParams.push({
+                  key: queryParamOverride,
+                  value: currentQueryParams[queryParamOverride],
```

Comment:
Simply making the URL a reflection of the feature flag state would be easier to do and thus would simplify the code. I was previously under the impression that reflecting `localStorage` was considered undesirable.

Reply:
Having the URL reflect the localStorage overrides would be a good reminder to the user that they are in some special state.

It will help us quickly troubleshoot problems like "why can't I see feature A after feature A has launched?" when user forgot they set enableFeatureA=false in local storage.  I am certain this will happen all the time.

Reply:
This makes lots of sense to me. That also ensures that sharing a link will result in the other person seeing the same thing. I've gone ahead and made this change and will be pushing a new commit quite soon.